config/setupMPI.cmake

```diff
@@ -399,7 +399,12 @@ macro( setupSpectrumMPI )
       # 1 resource set; OMP_NUM_THREADS tasks; no gpu; no binding
       set( MPIEXEC_OMP_PREFLAGS "--nrs 1 -c $ENV{OMP_NUM_THREADS} -g 0 --bind none")
     elseif( MPIEXEC_EXECUTABLE MATCHES lrun )
-      set( MPIEXEC_OMP_PREFLAGS "-c $ENV{OMP_NUM_THREADS} --threads=$ENV{OMP_NUM_THREADS} --bind=off -v" )
+      if( DEFINED ENV{OMP_NUM_THREADS} )
+        # --bind=off
+        set( MPIEXEC_OMP_PREFLAGS "--pack --threads=$ENV{OMP_NUM_THREADS} -c$ENV{OMP_NUM_THREADS} -v" )
+      else()
+        set( MPIEXEC_OMP_PREFLAGS "--pack --threads=4 -c 4 -v" )
```

Comment:
Is this `--threads=4` used by pthreads in the worker model?

Reply:
If you don't specify this option, lrun generates warnings about using more threads that requested/allowed.  I think this is the default of 4 hw threads per core.  I would like your input as all of these options are a bit arcane to me.  The hard part is creating commands that allow us to pack a node.  Prior to these changes smpi was running rank 0 of all mpi jobs on core 0.  That was very bad.

Reply:
OK, I think I get what's happening, thanks.